src/core/net/srp_client.cpp

```diff
@@ -403,6 +406,11 @@ Error Client::Start(const Ip6::SockAddr &aServerSockAddr, Requester aRequester)
 #endif
 
 exit:
+    if(error != kErrorNone && mSocket.IsOpen())
+    {
+        LogInfo("cannot connect to server %s port %d: %s", aServerSockAddr.GetAddress().ToString().AsCString(), aServerSockAddr.GetPort(), strerror(errno));
+        IgnoreError(mSocket.Close());
+    }
```

Comment:
The change sounds good, but I suggest some changes to the code:

- Let's move this check closer to the `Connect()` call.
- We cannot use `errno` as this is POSIX-specific and OT code can run on various platforms/OSes.
  - Even on POSIX platforms, it is applicable only when `OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE` is enabled and used (i.e., UDP sockets are delegated to the platform layer, and POSIX APIs are used for opening/connecting/closing, etc.).
- No need to log the IPv6 address and port separately. Since `aServerSockAddr` is a `SockAddr`, we can directly use `ToString()`, which will include both the IPv6 address and port.

Here is my suggestion, after `Open()`:

```c++

    SuccessOrExit(error = mSocket.Open());

    error = mSocket.Connect(aServerSockAddr);

    if (error != kErrorNone)
    {
        LogInfo("Failed to connect to server %s: %s", aServerSockAddr.ToString().AsCString(), ErrorToString(error));
        IgnoreError(mSocket.Close());
        ExitNow();
     }

    LogInfo("%starting, server %s", (aRequester == kRequesterUser) ? "S" : "Auto-s",
            aServerSockAddr.ToString().AsCString());
```

Reply:
ok, works for me, thanks.
I'll do & test the changes and update de PR